src/plugins/customBorders/customBorders.js

```diff
@@ -144,40 +143,39 @@ class CustomBorders extends BasePlugin {
     * @param {Object} borderObject Object with `top`, `right`, `bottom` and `left` properties.
     */
   setBorders(selection, borderObject) {
-    arrayEach(selection, (cell) => {
-      if (!Array.isArray(cell)) {
-        const topLeft = cell.getTopLeftCorner();
-        const bottomRight = cell.getBottomRightCorner();
+    const defaultBorderKeys = ['top', 'right', 'bottom', 'left'];
+    const borderKeys = borderObject ? Object.keys(borderObject) : defaultBorderKeys;
 
-        rangeEach(topLeft.row, bottomRight.row, (row) => {
-          rangeEach(topLeft.col, bottomRight.col, (column) => {
-            this.removeAllBorders(row, column);
-          });
-        });
-
-        this.prepareBorderFromCustomAddedRange(cell, borderObject);
-
-      } else {
+    arrayEach(selection, (cell) => {
+      if (Array.isArray(cell)) {
```

Comment:
To detect selection type I can suggest to you to use `normalizeSelectionFactory` function from `selection/utils.js` file. If you agree with me to use this helper please consider re-export that helpers in `selection/index.js` file.

Reply:
Done.